### ISSUE TITLE
Zig emits `NUL` file when compiling on Windows

### ISSUE THREAD

**Issue by GoNZooo** [ziglang/zig#5989]
When building the current latest `zls` (`fdba27b6e0feb13071ef40f0c06f6c0263e4498d`) I stumbled on the following compiler error and accompanying odd result:

```
❯ zig version
0.6.0+cf4936bcb
❯ zig build
Semantic Analysis [485/1631] Assertion failed at D:\a\1\s\src\os.cpp:1931 in slice_to_prefixed_file_w. This is a bug in the Zig compiler.
Unable to dump stack trace: debug info stripped
zls...The following command exited with error code 3:
C:\Users\ricka\tools\zig\zig-windows-x86_64-0.6.0+cf4936bcb\zig.exe build-exe C:\Users\ricka\code\zig\zls\src\main.zig --pkg-begin build_options C:\Users\ricka\code\zig\zls\zig-cache\zls_build_options.zig --pkg-end --cache-dir C:\Users\ricka\code\zig\zls\zig-cache --name zls --pkg-begin known-folders C:\Users\ricka\code\zig\zls\src\known-folders\known-folders.zig --pkg-end --cache on

Build failed. The following command failed:
C:\Users\ricka\code\zig\zls\zig-cache\o\WFdL0_Tr78-0OOUfYOT_1YFMqjwkG3NupnGAS31ahTNVzoaSSo1tM_SYCY3fcwvz\build.exe C:\Users\ricka\tools\zig\zig-windows-x86_64-0.6.0+cf4936bcb\zig.exe C:\Users\ricka\code\zig\zls C:\Users\ricka\code\zig\zls\zig-cache
```

The compiler seems to write a `NUL` file into the directory which needs something like the following to remove:

```
> cmd
> rename \\.\C:\path\to\NUL begone
> del C:\path\to\begone
```

**Comment by squeek502** [comment]
The assert in question:

https://github.com/ziglang/zig/blob/cf4936bcb0d48df5578efe76156c5c7ffc0d7b1e/src/os.cpp#L1929-L1932

**Comment by GoNZooo** [comment]
As an update to this, zig now outputs this `NUL` file regardless of errors being present or not. I hadn't noticed before because I downgraded to accommodate for `zls` previously and stayed there. I at least have a `NUL` file in my directory even on success, meaning it has nothing to do with the assertion.

**Comment by squeek502** [comment]
Could you provide a minimal test case / example if possible? Is this specific to `zls`'s setup or is it more general?

**Comment by GoNZooo** [comment]
I've noticed it happening in a completely unrelated repo now and it seems like a general compiler bug and not necessarily related to zls specifics. I'm trying to figure out what the commonalities are between these cases.

**Comment by code-disaster** [comment]
I suspect #5960. In child_process.zig in the spawnWindows() function, "NUL" is opened with `windows.OpenFile()`, but closed with `os.close()`. If I replace the latter with `windows.CloseHandle()`, the file seems to get cleaned up properly.

**Comment by deltaphc** [comment]
I've observed this as well. The "NUL" file it leaves behind cannot be deleted through File Explorer; it asks for admin permission and the file _seems_ to disappear, but when you leave and come back to the folder, NUL is still there. You have to through the extra steps mentioned in the OP in order to actually delete it.

**Comment by BarabasGitHub** [comment]
I'm also having this issue (for a while now, only just investigated a bit more). It seems to only happen for me when I have a call to `linkSystemLibrary` in my `build.zig` with anything other than `"c"`. I don't know why or how, but maybe it'll help someone find the issue. 

To clean up the nul file you can do: `del \\.\C:\full\path\to\NUL`

**Comment by FireFox317** [comment]
Seems like this has been solved in a8a806e925ddb2362e0021bd83dd73f7389a2dbb, so this can be closed? I did not verify on windows.

**Comment by daurnimator** [comment]
Fixed via #6263
